### PR TITLE
add auth cex account data

### PR DESCRIPTION
**Pull request by NNagato**

### CODE REVIEW

File: exchange/binance/binance_endpoint.go [hunk 1]
```diff
@@ -36,6 +37,7 @@ type Endpoint struct {
 	marketDataBaseURL  string
 	accountDataBaseURL string
 	accountID          string
+	ah                 *authhttp.AuthHTTP
```
Comment (halink0803) [inline exchange/binance/binance_endpoint.go:40]: why don't you put this field a meaningful name? `ah` seems do not fine for me.

File: exchange/binance/binance_endpoint.go [hunk 2]
```diff
@@ -159,13 +161,8 @@ func (ep *Endpoint) Trade(tradeType string, pair commonv3.TradingPairSymbols, ra
...
 		"price":       strconv.FormatFloat(rate, 'f', -1, 64),
 	}
-	respBody, err := ep.GetResponse(
-		"POST",
-		fmt.Sprintf("%s/api/v3/order/%s", ep.accountDataBaseURL, ep.accountID),
-		params,
-		false,
-		common.NowInMillis(),
-	)
+	respBody, err := ep.ah.DoReq(fmt.Sprintf("%s/api/v3/order/%s", ep.accountDataBaseURL, ep.accountID),
```
Comment (halink0803) [inline exchange/binance/binance_endpoint.go:164]: separate lines make it looks clearer.

File: exchange/binance/binance_endpoint.go [hunk 3]
```diff
@@ -225,16 +222,11 @@ func (ep *Endpoint) GetAccountTradeHistory(
 // WithdrawHistory get withdraw history from binance
 func (ep *Endpoint) WithdrawHistory(startTime, endTime uint64) (exchange.Binawithdrawals, error) {
 	result := exchange.Binawithdrawals{}
-	respBody, err := ep.GetResponse(
-		http.MethodGet,
-		fmt.Sprintf("%s/wapi/v3/withdrawHistory/%s", ep.accountDataBaseURL, ep.accountID),
-		map[string]string{
+	respBody, err := ep.ah.DoReq(fmt.Sprintf("%s/wapi/v3/withdrawHistory/%s", ep.accountDataBaseURL, ep.accountID),
```
Comment (halink0803) [inline exchange/binance/binance_endpoint.go:225]: same here.

File: lib/auth-http/auth_http.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+	switch httpMethod {
+	case http.MethodPost, http.MethodPut:
+		req.Header.Add("Content-Type", "application/json")
+	case http.MethodGet:
+		q := req.URL.Query()
+		for k, v := range params {
+			q.Add(k, v)
+		}
+		fmt.Println(q.Encode())
```
Comment (halink0803) [inline lib/auth-http/auth_http.go:65]: Is this for debug purpose. Remove it or use logger for it.

File: lib/auth-http/auth_http.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+		req.URL.RawQuery = q.Encode()
+
+		fmt.Println(req.URL.String())
```
Comment (halink0803) [inline lib/auth-http/auth_http.go:68]: same here.

**Review by halink0803 — CHANGES_REQUESTED** [review]
There are some lines need to be changed.

**Review by halink0803 — APPROVED** [review]
LGTM

**Comment by secmask** [comment]
@NNagato how can we turn off auth when we don't want to use it?

**Comment by NNagato** [comment]
> @NNagato how can we turn off auth when we don't want to use it?

just pass empty access keys and the request will be executed normally

**Pull request merged**